### PR TITLE
Refactor the code so that adding fields to Person is modular.

### PR DESCRIPTION
**Pull request by TypeDefinition**
The current code base is highly coupled, and adding a field to Person will result in changes in an unreasonable number of files.

A more modular system will increase clarity and make the system much more maintainable then the utter piece of crap it currently is.

Let's make the system more modular.

### CODE REVIEW

**Review by LapisRaider — APPROVED** [review]
LGTM, just some very very minor changes that should not affect the behaviour at all.

**Review by KeithCZW — APPROVED** [review]
Don't see any glaring syntax issues.
I assume that the test cases work.

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -39,34 +33,38 @@
...
-            + "[" + PREFIX_PHONE + "PHONE] "
-            + "[" + PREFIX_EMAIL + "EMAIL] "
-            + "[" + PREFIX_ADDRESS + "ADDRESS] "
-            + "[" + PREFIX_REMARK + "ADDRESS] "
-            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "[" + Name.PREFIX + "NAME] "
+            + "[" + Phone.PREFIX + "PHONE] "
+            + "[" + Email.PREFIX + "EMAIL] "
+            + "[" + Address.PREFIX + "ADDRESS] "
+            + "[" + Remark.PREFIX + "ADDRESS] "
```
[the remark this answers is not in the document]
Comment (TypeDefinition) [reply src/main/java/seedu/address/logic/commands/EditCommand.java]: Changed according to review.

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 2]
```diff
@@ -44,30 +36,12 @@ public Command parseCommand(String userInput) throws ParseException {
...
-        case RemarkCommand.COMMAND_WORD:
-            return new RemarkCommandParser().parse(arguments);
-        default:
-            throw new ParseException(MESSAGE_UNKNOWN_COMMAND);
+
+        if (CommandRegistry.PARSERS.containsKey(commandWord)) {
+            return CommandRegistry.PARSERS.get(commandWord).parse(arguments);
         }
+
+        throw new ParseException(MESSAGE_UNKNOWN_COMMAND);
```
[the remark this answers is not in the document]
Comment (TypeDefinition) [reply src/main/java/seedu/address/logic/parser/AddressBookParser.java]: Changed according to review.

File: src/main/java/seedu/address/logic/parser/FieldParser.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package seedu.address.logic.parser;
+
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Field;
+
+public interface FieldParser<T extends Field> {
```
[the remark this answers is not in the document]
Comment (TypeDefinition) [reply src/main/java/seedu/address/logic/parser/FieldParser.java:9]: Changed according to review.

File: src/main/java/seedu/address/logic/parser/NameParser.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Name;
+
+public class NameParser {
```
[the remark this answers is not in the document]
Comment (TypeDefinition) [reply src/main/java/seedu/address/logic/parser/NameParser.java:11]: Changed according to review.

File: src/main/java/seedu/address/logic/parser/RemarkParser.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Remark;
+
+public class RemarkParser {
```
[the remark this answers is not in the document]
Comment (TypeDefinition) [reply src/main/java/seedu/address/logic/parser/RemarkParser.java:8]: Changed according to review.

**Pull request merged**